packages/hawtio/src/ui/page/HawtioHeader.tsx

```diff
@@ -104,9 +105,27 @@ const HawtioHeaderToolbar: React.FunctionComponent = () => {
     userItems.pop()
   }
 
+  /*
+   * Determine which plugin is currently displaying
+   * based on the path of the current location
+   */
+  const pluginFromLocation = (): Plugin | null => {
+    const path = location.pathname
+    if (!path) return null
+
+    return plugins.find(plugin => plugin.path === path) || null
+  }
+
+  const plugin = pluginFromLocation()
+
   return (
     <Toolbar id='hawtio-header-toolbar'>
       <ToolbarContent>
+        <ToolbarGroup>
+          {plugin?.toolbarItems?.map((comp, index) => (
+            <ToolbarItem key={`custom-toolbar-item-${index}`}>{React.createElement(comp)}</ToolbarItem>
```

Comment:
I would name the key as `hawtio-header-toolbar-plugin-item-${index}`. I want the ids to be named hierarichally starting from the name of the top-level component (i.e. `hawtio-header` in this case) so that the ids are named consistently throughout the entire console.